src/execution/subscribe.ts

```diff
@@ -51,17 +51,17 @@ export async function subscribe(
 ): Promise<AsyncGenerator<ExecutionResult, void, void> | ExecutionResult> {
   // If a valid execution context cannot be created due to incorrect arguments,
   // a "Response" with only errors is returned.
-  let exeContext: ExecutionContext;
+  let executor: Executor;
   try {
-    exeContext = buildExecutionContext(args);
+    executor = new Executor(args);
```

Comment:
That was the initial approach, but @IvanGoncharov feels strongly that rather than the mainline `execute` function taking a arguments that allow you to specify the type of `executor` to use, those who wish to use a custom `executor` should also write their own custom `execute` wrapper that calls the custom `executor`. This means that all frameworks, servers, etc, should really allow users to pass a custom execute function. That's the way of the future anyways, as pointed to by Envelop, graphql-helix, etc.

See https://github.com/graphql/graphql-js/pull/3163#issuecomment-859546546

Reply:
Although maybe your suggestion has to do specifically with the subscribe function? I think another PR would make sense to allow one of the `subscribe` arguments be the `execute` function to use, but I think that's outside the scope of this initial change, and would be a good idea even if the `Executor` class concept never gets merged.